### PR TITLE
Replicate infrastructure with terraform

### PR DESCRIPTION
**Pull request by sgibson91**
This PR is attempting to recreate our infrastructure using terraform so that deployment is more reproducible.

### CODE REVIEW

**Comment by sgibson91** [comment]
@OscartGiles would you mind reviewing please?

**Comment by OscartGiles** [comment]
@sgibson91 - No problem. I'll take a look today or tomorrow.

File: terraform/main.tf [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+    }
+}
+
+# Key Vault
+resource "azurerm_key_vault" "keyvault" {
+    name                       = "hub23-keyvault"
+    location                   = azurerm_resource_group.rg.location
+    resource_group_name        = azurerm_resource_group.rg.name
+    sku_name                   = "standard"
+    tenant_id                  = data.azurerm_subscription.current.tenant_id
```
Comment (OscartGiles) [inline terraform/main.tf:85]: @jemrobinson Might be useful for future reference. I think we stored this in a key vault.

File: terraform/main.tf [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+    soft_delete_enabled        = true
+    soft_delete_retention_days = 90
+}
+
+# Kubernetes Cluster
+resource "azurerm_kubernetes_cluster" "k8s" {
+    name                = "hub23cluster"
+    kubernetes_version  = "1.16.15"
```
Comment (OscartGiles) [inline terraform/main.tf:93]: I think version 1.16.x will stop being supported on AKS in Jan 2021. Might be worth upgrading here?
I found this stopped me being able to add new node pools etc... and eventually Azure might upgrade it for you (https://docs.microsoft.com/en-us/azure/aks/supported-kubernetes-versions)

**Review by OscartGiles — APPROVED** [review]
Looks sensible to me. Are you using the local backend?
We used Azure (basically just stores the state file in blob storage). 
But as mentioned before there's now [terraform cloud](https://www.terraform.io/docs/cloud/migrate/index.html) to do something similar (haven't tried it out).

**Comment by sgibson91** [comment]
Yes, I'm using local backend right now. This isn't really a collaborative repo so it's not _super_ important.

**Pull request merged**